src/managers/RoleManager.js

```diff
@@ -218,6 +218,24 @@ class RoleManager extends CachedManager {
     return clone;
   }
 
+  /**
+   * Deletes a role.
+   * @param {RoleResolvable} role The role to delete
+   * @param {string} [reason] Reason for deleting the role
+   * @returns {Promise<Role>}
+   * @example
+   * // Delete a role
+   * guild.roles.delete('222079219327434752', 'The role needed to go')
+   *   .then(deleted => console.log(`Deleted role ${deleted.name}`))
+   *   .catch(console.error);
+   */
+  async delete(role, reason) {
+    const resolvedRole = this.resolve(role);
+    await this.client.api.guilds[this.guild.id].roles[resolvedRole.id].delete({ reason });
+    this.client.actions.GuildRoleDelete.handle({ guild_id: this.guild.id, role_id: resolvedRole.id });
+    return resolvedRole;
```

Comment:
We can just do what `MessageManager#delete` (and possibly other places as well) do: return void so it doesn't rely on caching.

https://github.com/discordjs/discord.js/blob/ee91f5a19cdcf15f17a80fc9a25a382b7cbcde9f/src/managers/MessageManager.js#L206-L211

Please revert to `resolveId`.

```suggestion
    const resolvedId = this.resolveId(role);
    await this.client.api.guilds[this.guild.id].roles[resolvedId].delete({ reason });
    this.client.actions.GuildRoleDelete.handle({ guild_id: this.guild.id, role_id: resolvedId });
```

Also update JSDocs to match those changes.

Reply:
This looks better! I have done this and updated the relevant typings.